app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -288,16 +288,29 @@ class NavigationOverlayFragment : Fragment() {
                         fxaButton.contentDescription =
                             resources.getString(R.string.fxa_navigation_item_new,
                                 resources.getString(R.string.app_name))
+                        resetFxaOnboardingShown()
                     }
                     AccountState.NeedsReauthentication -> {
                         fxaButton.setImageResource(R.drawable.ic_fxa_needs_reauthentication)
                         fxaButton.contentDescription =
                             resources.getString(R.string.fxa_navigation_item_sign_in_again)
+                        resetFxaOnboardingShown()
                     }
                 }
             }
     }
 
+    /**
+     * We need to reset the state of FxA onboarding screen to ensure that we do not show onboarding on every startup (#2861).
```

Comment:
nit: the first sentence of a kdoc should be a brief summary of what the thing does. The reason for this is that this one-line summary is used by the kdoc generation tools (or at least it was for javadoc) to provide a summary and clicking on it will provide the extended text.

Not that we generate kdoc but it's good to follow convention and, if we did ever generate kdoc, it could be useful.